ql/methods/montecarlo/fractionalpathgenerator.hpp

```diff
@@ -0,0 +1,131 @@
+/* -*- mode: c++; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- */
+
+/*
+ Copyright (C) 2022 Chester Wong
+ Copyright (C) 2022 Wojciech Czernous
+
+ This file is part of QuantLib, a free-software/open-source library
+ for financial quantitative analysts and developers - http://quantlib.org/
+
+ QuantLib is free software: you can redistribute it and/or modify it
+ under the terms of the QuantLib license.  You should have received a
+ copy of the license along with this program; if not, please email
+ <quantlib-dev@lists.sf.net>. The license is also available online at
+ <http://quantlib.org/license.shtml>.
+
+ This program is distributed in the hope that it will be useful, but WITHOUT
+ ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ FOR A PARTICULAR PURPOSE.  See the license for more details.
+*/
+
+/*! \file fractionalpathgenerator.hpp
+    \brief Generates random paths using a sequence generator
+*/
+
+#ifndef quantlib_montecarlo_fractional_path_generator_hpp
+#define quantlib_montecarlo_fractional_path_generator_hpp
+
+#include "path.hpp"
+#include <ql/math/randomnumbers/fractionalnoisersg.hpp>
+
+using std::pow;
+
+namespace QuantLib {
+    //! Generates random rough paths using a sequence generator
+    /*! Generates random rough paths with Hurst parameter H
+        using a gaussian sequence generator
+
+        \ingroup mcarlo
+
+        \test see the test for the underlying FractionalGaussianNoiseRsg
+    */
+    template <class GSG>
+    // It could be a subtype of FractionalPathGenerator<GSG>, 
```

Comment:
I think it would be helpful to add an example to the `Examples` directory to demonstrate how to use this new kind of path generator in a monte carlo simulation.